src/main/resources/avro/reads.avdl

```diff
@@ -177,7 +177,38 @@ record LinearAlignment {
 }
 
 /**
-Each read alignment describes a linear alignment with additional information
+A string-to-reference-graph alignment can be represented by one CIGAR string and
+one `Path` through multiple `Reference`s, against which the CIGAR string is
+interpreted.
+
+Note that `Path`s are restricted to following reference adjacencies.
+If you want to align your read following adjacencies that are
+not present in the `ReferenceSet` to which you are aligning, you have a
+"chimeric" alignment, and should use multiple `ReadAlignment`s and the
+supplementaryAlignment flag instead of a single `GraphAlignment`.
+
+Some especially large deletions could be represented just as well as a large
+deletion in the CIGAR string, or as a chimeric alignment.
+*/
+record GraphAlignment {
```

Comment:
On Wed, Jan 21, 2015 at 9:30 PM, CH Albach <notifications@github.com> wrote:

> In src/main/resources/avro/reads.avdl
> <https://github.com/ga4gh/schemas/pull/198#discussion_r23356303>:
>
> > @@ -177,7 +177,38 @@ record LinearAlignment {
> >  }
> >
> >  /**
> > -Each read alignment describes a linear alignment with additional information
> > +A string-to-reference-graph alignment can be represented by one CIGAR string and
> > +one `Path` through multiple `Reference`s, against which the CIGAR string is
> > +interpreted.
> > +
> > +Note that `Path`s are restricted to following reference adjacencies.
> > +If you want to align your read following adjacencies that are
> > +not present in the `ReferenceSet` to which you are aligning, you have a
> > +"chimeric" alignment, and should use multiple `ReadAlignment`s and the
> > +supplementaryAlignment flag instead of a single `GraphAlignment`.
> > +
> > +Some especially large deletions could be represented just as well as a large
> > +deletion in the CIGAR string, or as a chimeric alignment.
> > +*/
> > +record GraphAlignment {
>
> The current PR allows one to define a GraphAlignment which takes a Path
> through a non-reference (variant) segment. Is this desired behavior, or an
> unintended side-effect of using Segment for both References and Variants
>
From my reading of the paper, and the comments in this PR, my understanding
> is that in a Graph-based model, Reads (for example) would still be aligned
> to a reference set, and not to a reference set + variant set(s).
>

Good question. I can imagine there are times when you want to align a read
to a segment not in the reference set - for example for reads that are
otherwise not mappable, and where the extra segment serves the purpose of
extending the reference. IMO, either we should (1) make it a lightweight
operation to extend a reference set to include such extra segments, (2)
permit the case you point out, where you can align to a segment in a
variant set - which is a bit odd (I think I'm -1 on this), or (3) (probably
preferred, so 1+) allow extra segments to be included with the read
alignments, as we do for variant sets, but for reads.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ga4gh/schemas/pull/198/files#r23356303>.
>